tests/unit/Cli/ConsoleTest.php

```diff
@@ -0,0 +1,490 @@
+                    function() {
+                        return "data";
+                    }
+                );
+
+                $console = new Console();
+                $console->setDI($di);
+                $dispatcher = $console->getDI()->getShared('dispatcher');
+
+                $console->handle([]);
+                expect($dispatcher->getTaskName())->equals('main');
+                expect($dispatcher->getActionName())->equals('main');
+                expect($dispatcher->getParams())->equals([]);
+                expect($dispatcher->getReturnedValue())->equals('mainAction');
+
+                $console->handle(
+                    [
+                        'task' => 'echo',
+                    ]
+                );
+
+                expect($dispatcher->getTaskName())->equals('echo');
+                expect($dispatcher->getActionName())->equals('main');
+                expect($dispatcher->getParams())->equals([]);
+                expect($dispatcher->getReturnedValue())->equals('echoMainAction');
+
+                $console->handle(
+                    [
+                        'task' => 'main',
+                        'action' => 'hello'
+                    ]
+                );
+                expect($dispatcher->getTaskName())->equals('main');
+                expect($dispatcher->getActionName())->equals('hello');
+                expect($dispatcher->getParams())->equals([]);
+                expect($dispatcher->getReturnedValue())->equals('Hello !');
+
+                $console->handle(
+                    [
+                        'task' => 'main',
+                        'action' => 'hello',
+                        'World',
+                        '######'
+                    ]
+                );
+                expect($dispatcher->getTaskName())->equals('main');
+                expect($dispatcher->getActionName())->equals('hello');
+                expect($dispatcher->getParams())->equals(array('World', '######'));
+                expect($dispatcher->getReturnedValue())->equals('Hello World######');
+
+                // testing module
+                try {
+                    $console->handle(
+                        [
+                            'module' => 'devtools',
+                            'task' => 'main',
+                            'action' => 'hello',
+                            'World',
+                            '######'
+                        ]
+                    );
+                    expect($dispatcher->getTaskName())->equals('main');
+                    expect($dispatcher->getActionName())->equals('hello');
+                    expect($dispatcher->getParams())->equals(array('World', '######'));
+                    expect($dispatcher->getReturnedValue())->equals('Hello World######');
+                } catch (\Exception $e) {
+                    expect($e->getMessage())->equals("Module 'devtools' isn't registered in the console container");
```

Comment:
Split please this test to 2 tests: the testing module and the testing namespace.

Also note if exception will not be thrown, the test will pass. This is why you have to use something like this: https://github.com/Codeception/Specify#exceptions